App/src/App.tsx

```diff
@@ -19,13 +19,18 @@ import axios from "axios";
 import Dossiers from "./pages/dossier/Dossiers";
 import DisplayManageableGroups from "./pages/ManageableGroups";
 import EditCourse from "./pages/EditCourse";
+import AddingUserToGroup from "./pages/addUserToGroup";
+import RemovingUserFromGroup from "./pages/RemoveUserFromGroup";
 import DeleteCourse from "./pages/DeleteCourse";
 
 export const UserContext = createContext<User | null>(null);
 
 export function App() {
     const [user, setUser] = useState<User | null>(initializeUser());
     const [isLoggedIn, setIsLoggedIn] = useState<boolean>(user != null ? true : false);
+    const [isAdminorGroupMaster, setIsAdminorGroupMaster] = useState<boolean>(
+        user != null ? (user.roles.includes("Admin") || user.masteredGroups != null ? true : false) : false
```

Comment:
user.masteredGroup should never be null. You should check if it's empty or not. If it's not empty, it means the user is a group master.